Userland/Libraries/LibWeb/HTML/FormControlInfrastructure.cpp

```diff
@@ -175,4 +176,91 @@ WebIDL::ExceptionOr<Optional<Vector<XHR::FormDataEntry>>> construct_entry_list(J
+            }
+            TRY(builder.try_append(lexer.consume_until(is_any_of("\r\n"sv))));
+        }
+        return builder.to_string();
+    };
+
+    auto escape_line_feed_carriage_return_double_quote = [](StringView value) -> ErrorOr<String> {
+        StringBuilder builder;
+        GenericLexer lexer { value };
+        TRY(builder.try_append(lexer.consume_until(is_any_of("\r\n\""sv))));
+        while (!lexer.is_eof()) {
+            switch (lexer.peek()) {
+            case '\r':
+                TRY(builder.try_append("%0D"sv));
+                break;
+            case '\n':
+                TRY(builder.try_append("%0A"sv));
+                break;
+            case '\"':
+                TRY(builder.try_append("%22"sv));
+                break;
+            default:
+                VERIFY_NOT_REACHED();
+            }
+            lexer.ignore(1);
+            TRY(builder.try_append(lexer.consume_until(is_any_of("\r\n\""sv))));
+        }
+        return builder.to_string();
+    };
+
+    // 1. For each entry of entry list:
+    //    - Replace every occurrence of U+000D (CR) not followed by U+000A (LF), and every occurrence of U+000A (LF) not preceded by U+000D (CR), in entry's name, by a string consisting of a U+000D (CR) and U+000A (LF).
+    //    - If entry's value is not a File object, then replace every occurrence of U+000D (CR) not followed by U+000A (LF), and every occurrence of U+000A (LF) not preceded by U+000D (CR), in entry's value, by a string consisting of a U+000D (CR) and U+000A (LF).
+    // 2. Return the byte sequence resulting from encoding the entry list using the rules described by RFC 7578, Returning Values from Forms: multipart/form-data, given the following conditions: [RFC7578]
+    //    - Each entry in entry list is a field, the name of the entry is the field name and the value of the entry is the field value.
+    //    - The order of parts must be the same as the order of fields in entry list. Multiple entries with the same name must be treated as distinct fields.
+    //    - Field names, field values for non-file fields, and filenames for file fields, in the generated multipart/form-data resource must be set to the result of encoding the corresponding entry's name or value with encoding, converted to a byte sequence.
+    //    - For field names and filenames for file fields, the result of the encoding in the previous bullet point must be escaped by replacing any 0x0A (LF) bytes with the byte sequence `%0A`, 0x0D (CR) with `%0D` and 0x22 (") with `%22`. The user agent must not perform any other escapes.
+    //    - The parts of the generated multipart/form-data resource that correspond to non-file fields must not have a `Content-Type` header specified.
+    //    - The boundary used by the user agent in generating the return value of this algorithm is the multipart/form-data boundary string. (This value is used to generate the MIME type of the form submission payload generated by this algorithm.)
```

Comment:
Would be helpful to split this up and actually move individual lines to where we're doing the thing instead of one big comment, one big blob of code :)

Reply:
Split these up and improvised a bit since the steps are not easily incorporated into our implementation.